src/abstractdataframe/abstractdataframe.jl

```diff
@@ -197,14 +201,7 @@ eltypes(df)
 ```
 
 """
-function eltypes(df::AbstractDataFrame)
-    ncols = size(df, 2)
-    res = Array(Type, ncols)
-    for j in 1:ncols
-        res[j] = eltype(df[j])
-    end
-    return res
-end
+eltypes(df::AbstractDataFrame) = map!(eltype, Vector{Type}(size(df,2)), columns(df))
```

Comment:
`eltype(col)` has result type `Any`, and IIRC `eltype(col)::Type` doesn't help getting the type of array inferred automatically. Probably it would be as fast as `map!()`, I just wanted to have a version without any `Any` :)